project-keeper/src/test/java/com/exasol/projectkeeper/dependencyupdate/ChangesFileUpdaterTest.java

```diff
@@ -30,17 +36,162 @@ class ChangesFileUpdaterTest {
+                                "### cve-1 (cwe-1) in dependency `coordinates-1`", "description-1", "")), //
+                testCase(List.of("existing", "summary"), List.of(vulnerability(1)),
+                        List.of("existing", "summary", "", "This release fixes the following vulnerability:", "",
+                                "### cve-1 (cwe-1) in dependency `coordinates-1`", "description-1", "")), //
+                testCase(null, List.of(vulnerability(1, emptyList())),
+                        List.of("", "This release fixes the following vulnerability:", "",
+                                "### cve-1 (cwe-1) in dependency `coordinates-1`", "description-1", "")), //
+                testCase(null, List.of(vulnerability(1, List.of("link-1"))),
+                        List.of("", "This release fixes the following vulnerability:", "",
+                                "### cve-1 (cwe-1) in dependency `coordinates-1`", "description-1", "#### References",
+                                "* link-1", "")), //
+                testCase(null, List.of(vulnerability(1, List.of("link-1", "link-2"))),
+                        List.of("", "This release fixes the following vulnerability:", "",
+                                "### cve-1 (cwe-1) in dependency `coordinates-1`", "description-1", "#### References",
+                                "* link-1", "* link-2", "")), //
+                testCase(null, List.of(vulnerability(1), vulnerability(2)),
+                        List.of("", "This release fixes the following 2 vulnerabilities:", "",
+                                "### cve-1 (cwe-1) in dependency `coordinates-1`", "description-1", "", //
+                                "### cve-2 (cwe-2) in dependency `coordinates-2`", "description-2", "")) //
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("summary")
+    void summaryUpdated(final TestCase<List<String>> test) {
+        final ChangesFile changes = ChangesFile.builder().summary(test.currentText == null ? null
+                : ChangesFileSection.builder("## Summary").addLines(test.currentText).build()).build();
+        final List<String> updatedSummary = getUpdatedChanges(test.vulnerabilities, changes).getSummarySection() //
+                .orElseThrow().getContent();
+        assertThat(updatedSummary, equalTo(test.expected));
+    }
+
+    static Stream<Arguments> securitySection() {
+        final ChangesFileSection featureSection = section("## Features", "* Feature 1");
+        final ChangesFileSection bugfixSection = section("## Bugfixes", "* Bugfix 1");
+        final ChangesFileSection existingSecuritySection = securitySection("* Existing security fix");
+        return Stream.of( //
+                testCaseSecurity(null, List.of(vulnerability(1)),
+                        List.of(securitySection("", "* #1: Fixed vulnerability cve-1 in dependency `coordinates-1`",
```

Comment:
Could we refactor to have

```
private static ChangesFileSection securitySectionExistingFix(int numberOfFixes) {
  return securitySection("* Existing security fix", "* #1: Fixed vulnerability cve-1 in dependency `coordinates-1`", "");
}
```

```
private static ChangesFileSection securitySectionFixed(int numberOfFixes) {
  String[] a = List.of("", "* #1: Fixed vulnerability cve-1 in dependency `coordinates-1`");
  if(numberOfFixes < 2) {
    return securitySection(a);
  } else {
    return securitySection(a + List.of("* #2: Fixed vulnerability cve-2 in dependency `coordinates-2`", "");
  }
}

```

Reply:
I think this will make the test harder to read. Is it OK to omit this?